upload/library/Sidane/Threadmarks/Model/Threadmarks.php

```diff
@@ -3,6 +3,102 @@
 class Sidane_Threadmarks_Model_Threadmarks extends XenForo_Model
 {
 
+  public function getMenuLimit(array $thread, array $nodePermissions = null, array $viewingUser = null)
+  {
+    $this->standardizeViewingUserReferenceForNode($thread['node_id'], $viewingUser, $nodePermissions);
+    
+    $menulimit = XenForo_Permission::hasContentPermission($nodePermissions, 'permission_forum_sidane_tm_menu_limit'); 
```

Comment:
Testing locally and the menu has no threadmarks regardless of the defined menu limit.

I think this line should be:

$menulimit = XenForo_Permission::hasPermission($viewingUser['permissions'], 'forum', 'sidane_tm_menu_limit');